hail/python/hailtop/batch/job.py

```diff
@@ -343,6 +344,48 @@ def always_run(self, always_run: bool = True) -> 'Job':
         self._always_run = always_run
         return self
 
+    def regions(self, regions: Optional[Union[str, List[str]]]) -> 'Job':
+        """
+        Set the cloud regions a job can run in.
+
+        Notes
+        -----
+        Can only be used with the :class:`.backend.ServiceBackend`.
+
+        Warning
+        -------
+        Not setting this option or setting it to None may result in additional egress
+        fees on GCP if a job is run in a region that is different from where the data and
+        image reside.
+
+        Examples
+        --------
+
+        >>> b = Batch(backend=backend.ServiceBackend('test'))
+        >>> j = b.new_job()
+        >>> (j.regions(['us-central1'])
+        ...   .command(f'echo "hello"'))
+
+        Parameters
+        ----------
+        regions:
+            The cloud region(s) to run this job in. When using the Hail maintained Batch Service on GCP,
+            the available regions are "us-central1", "us-east1", "us-east4", "us-west1", "us-west2",
+            "us-west3", and "us-west4".
```

Comment:
I'm a bit hesitant to include this in the library documentation. The library documentation shouldn't be so specific to one particular instance of Hail Batch. We should point users at the GCP docs. We should also have an API endpoint that lists the supported regions which users can call like `hb.fetch_available_regions()` or something like that.

Reply:
I think the api endpoint is a good idea and I can add it to the Batch Service documentation page.